lib/plugin-loader.js

```diff
@@ -752,71 +756,275 @@ PluginLoader.prototype = {
+            } 
+            else if (acceptedOS.includes("!"+components[APP_SERVER_COMP_ID].os)) {
+              plugin.error = new Error(`(${components[APP_SERVER_COMP_ID].os}) is not a supported platform for the App server. Skipping (${plugin.identifier})... Supported: ${acceptedOS}`);
+            }
+            else {
+              console.log(`App server could not verify (${components[APP_SERVER_COMP_ID].os}) as a supported platform to install (${plugin.identifier}). Proceeding anyway...`);
+            }
+          }
+
+          if (acceptedCPU && acceptedCPU.length > 0 && components[APP_SERVER_COMP_ID].cpu) {
+            if (acceptedCPU.includes(components[APP_SERVER_COMP_ID].cpu)) {
+              console.log(`(${components[APP_SERVER_COMP_ID].cpu}) is a supported architecture for the App server to install (${plugin.identifier}).`);
+            } else if (acceptedCPU.includes("!"+components[APP_SERVER_COMP_ID].cpu)) {
+              plugin.error = new Error(`(${components[APP_SERVER_COMP_ID].cpu}) is not a supported architecture for the App server. Skipping (${plugin.identifier})... Supported: ${acceptedCPU}`);
+            }
+            else {
+              console.log(`App server could not verify (${components[APP_SERVER_COMP_ID].cpu}) as a supported architecture to install (${plugin.identifier}). Proceeding anyway...`);
+            }
+          }
         }
-        const pluginConfiguration = configService.getPluginConfiguration(
-          pluginDef.identifier, pluginDef.location,
-          this.options.serverConfig, this.options.productCode);
-          bootstrapLogger.debug("ZWED0165I", pluginDef.identifier, JSON.stringify(pluginConfiguration)); //bootstrapLogger.debug(`For plugin with id=${pluginDef.identifier}, internal config` 
-                                //+ ` found=\n${JSON.stringify(pluginConfiguration)}`);
-
-        const plugin = makePlugin(pluginDef, pluginConfiguration, pluginContext,
-                                  false, this.options.langManagers);
-        if (plugin) {
-          bootstrapLogger.info("ZWED0124I", plugin.identifier, plugin.location); //bootstrapLogger.log(bootstrapLogger.INFO,
-            //`Plugin ${plugin.identifier} at path=${plugin.location} loaded.\n`);
-            bootstrapLogger.debug("ZWED0164I", plugin.toString()); //bootstrapLogger.debug(' Content:\n' + plugin.toString());
-          let frozen = zluxUtil.deepFreeze(plugin);
-          this.plugins.push(frozen);
-          newPlugins.push(frozen);
-          this.pluginMap[plugin.identifier] = plugin;
-          successCount++;
-        } else {
-          bootstrapLogger.info("ZWED0125I", pluginDef.identifier); //bootstrapLogger.log(bootstrapLogger.INFO,
-            //`Plugin ${pluginDef.identifier} not loaded`);
-          this.pluginMap[pluginDef.identifier] = {}; //mark it as having failed so it will not be retried
+
+        if (acceptedComponents[AGENT_COMP_ID]) { /* --- Agent component --- */
```

Comment:
not seeing why you need a block per each component here. the claims are the same:
do os match
do cpu match
do version match
do endpoints match
and maybe more.
you could run through a loop rather than pasting a block per component to do these checks.